tools/wptrunner/wptrunner/browsers/firefox.py

```diff
@@ -870,6 +873,37 @@ def start(self, group_metadata, **kwargs):
         self.marionette_port = get_free_port()
         super().start(group_metadata, **kwargs)
 
+    def stop(self, force=False):
+        # Initially wait for any WebDriver session to cleanly shutdown
```

Comment:
Can you please explain what the benefits of these extra checks are? Maybe extend the comment to explain it in the code.